pandas/core/frame.py

```diff
@@ -450,7 +450,7 @@ def __init__(
 
         # For data is list-like, or Iterable (will consume into list)
         elif isinstance(data, abc.Iterable) and not isinstance(data, (str, bytes)):
-            if not isinstance(data, abc.Sequence):
+            if not isinstance(data, (abc.Sequence, Categorical)):
```

Comment:
Should this be any extension dtype? Not just categorical?

Reply:
@TomAugspurger 
Yes. you are right. Thank you for pointing out. So I change it. Also add test case too except "DatetimeArray" and "TimedeltaArray". Because these are currently experimental and also these have no reshape method so can't use "assert_frame_equal".